src/ui/Pager/Pager.ts

```diff
@@ -139,6 +141,8 @@ export class Pager extends Component {
       this.handleQueryStateModelChanged(data)
     );
 
+    this.bind.onRootElement(ResultListEvents.newResultsDisplayed, () => this.hideIfResultListHasInfiniteScrollEnabled());
```

Comment:
Why are we doing this on "every new result displayed" ?
Isn't that extremely wasteful ?

At the very list I'd do it only on "querySuccess" maybe ?